src/api/attestation_store.rs

```diff
@@ -0,0 +1,110 @@
+use alloc::string::String;
+use alloc::vec::Vec;
+use persistent_store::{StoreError, StoreUpdate};
+
+use crate::env::Env;
+
+/// Identifies an attestation.
+pub enum Id {
+    Batch,
+    Enterprise { rp_id: String },
+}
+
+#[cfg_attr(feature = "std", derive(Debug, PartialEq, Eq))]
+pub struct Attestation {
+    /// ECDSA private key (big-endian).
+    pub private_key: [u8; 32],
+    pub certificate: Vec<u8>,
+}
+
+/// Stores enterprise or batch attestations.
+///
+/// Implementations don't need to distinguish different attestations. In particular, setting one
+/// attestation may set other ones.
```

Comment:
Actually, I think we want to keep the comment. Because enterprise attestation may be shared between different RP ids. If we want to improve we could add a function to know when this happens:

```rust
/// Returns whether 2 attestation ids share the same attestation.
fn shared_attestation(&self, a: &Id, b: &Id) -> Result<bool, Error>;
```

But since we won't use it in the CTAP library, I'm not sure it's worth doing it now.

Reply:
This is resolved by removing the RP ID string I think.